modules/formulize/class/graphScreen.php

```diff
@@ -0,0 +1,128 @@
+        } else {
+            $sql = sprintf("UPDATE %s SET width = %u, height = %u, orientation = %s, bgr = %u, bgg = %u, bgb = %u, barr = %u, barg = %u, barb = %u, ops = %s, labelelem = %u, dataelem = %u, defaultview = %s, limitviews = %s, usecurrentviewlist = %s WHERE sid = %u", $this->db->prefix('formulize_screen_graph'), $screen->getVar('width'), $screen->getVar('height'), $this->db->quoteString($screen->getVar('orientation')), $screen->getVar('bgr'), $screen->getVar('bgg'), $screen->getVar('bgb'), $screen->getVar('barr'), $screen->getVar('barg'), $screen->getVar('barb'), $this->db->quoteString($screen->getVar('ops')), $screen->getVar('labelelem'), $screen->getVar('dataelem'), $this->db->quoteString($screen->getVar('defaultview')), $this->db->quoteString(serialize($screen->getVar('limitviews'))), $this->db->quoteString($screen->getVar('usecurrentviewlist')), $screen->getVar('sid'));
+        }
+        $result = $this->db->query($sql);
+        if (!$result) {
+            print "Error: could not save the screen properly: ".mysql_error()." for query: $sql";
+            return false;
+        }
+        return $sid;
+	}
+
+	// 	THIS METHOD MIGHT BE MOVED UP A LEVEL TO THE PARENT CLASS
+	function get($sid) {
+		$sid = intval($sid);
+		if ($sid > 0) {
+			$sql = 'SELECT * FROM '.$this->db->prefix('formulize_screen').' AS t1, '. $this->db->prefix('formulize_screen_graph').' AS t2 WHERE t1.sid='.$sid.' AND t1.sid=t2.sid';
+			if (!$result = $this->db->query($sql)) {
+				return false;
+			}
+			$numrows = $this->db->getRowsNum($result);
+			if ($numrows == 1) {
+				$screen = new formulizeGraphScreen();
+				$screen->assignVars($this->db->fetchArray($result));
+				return $screen;
+			}
+		}
+		return false;
+
+	}
+
+	// THIS METHOD HANDLES ALL THE LOGIC ABOUT HOW TO ACTUALLY DISPLAY THIS TYPE OF SCREEN
+	// $screen is a screen object
+	function render($screen, $entry, $settings = "") { // $settings is used internally to pass list of entries settings back and forth to editing screens
+		$bgc = array(
+			"R" => $screen->getVar('bgr'),
+			"G" => $screen->getVar('bgg'),
+			"B" => $screen->getVar('bgb')
+		);
+		$barc = array(
+			"R" => $screen->getVar('barr'),
+			"G" => $screen->getVar('barg'),
+			"B" => $screen->getVar('barb')
+		);
+		$options = array(
+			"width" => $screen->getVar('width'),
+			"height" => $screen->getVar('height'),
+			"orientation" => $screen->getVar('orientation'),
+			"backgroundcolor" => $bgc,
+			"barcolor" => $barc,
+			"defaultview" => $screen->getVar('defaultview'),
+			"limitviews" => $screen->getVar('limitviews'),
+			"usecurrentviewlist" => $screen->getVar('usecurrentviewlist')
+		);
+		include_once XOOPS_ROOT_PATH."/modules/formulize/include/graphdisplay.php";
+		displayGraph('Bar', $screen->getVar('fid'), $screen->getVar('frid'), $screen->getVar('labelelem'), $screen->getVar('dataelem'), $screen->getVar('ops'), $options);
```

Comment:
There's a problem here. These two variables are integers, which represent element IDs: $screen->getVar('labelelem'), $screen->getVar('dataelem'). Later in graphdislay.php, these are used like this: display($entry, $dataElement);. It works if the element has the same ID and handle, but if the element handle has been changed to text, such as, 'client_name' then it fails. The code in one of these places should be changed so that it's consistently using element IDs or handles, and it's probably better to use handles since that's what a lot of the other code does.